### ISSUE TITLE
no-unnecessary-type-assertion relies on no-unused-variable to work

### ISSUE THREAD

**Issue by mxchange** [palantir/tslint#3471]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI (./node_modules/tslint/bin/tslint --config tslint.json)

#### TypeScript code being linted

```ts
this.user = this.linkedUserForm.get('user') as FormControl;
```

with `tslint.json` configuration:

```json
"no-unused-variable": true,
"no-unnecessary-type-assertion": true,
```

#### Actual behavior
Will only warn for unnecessary type assertion when `no-unused-variable` is on. 
But in my view, is wrongly warning about the unessecity of the type assertion, because if I don't do `as FormControl` at-loader will complain about it: 
``` Type 'AbstractControl | null' is not assignable to type 'FormControl'.
  Type 'null' is not assignable to type 'FormControl'.
```
While that might be `at-loader`s fault, I think it is probably correct in warning about it.
But more importantly I think the reliance on `no-unused-variable` is wrong.

#### Expected behavior
The warnings to throw when `no-unused-variable` is _not_ used.
And perhaps not complain about  the above code being a wrong assertion.

**Comment by ajafff** [comment]
`no-unused-variable` does not enable `no-unnecessary-type-assertion`. It messes with the type information and therefore breaks other rules that use type information.
As you already noted above, there should be no error in your code.

Closing as a duplicate of #2736
